source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -71,6 +69,27 @@ FaultFilter::~FaultFilter() { ASSERT(!delay_timer_); }
 // if we inject a delay, then we will inject the abort in the delay timer
 // callback.
 Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
+  // Route-level configuration overrides filter-level configuration
+  // TODO (rshriram): We should not be using Runtimes when reading from route-level
+  // faults until we parameterize the runtime keys for faults based on the route name.
+  // Its okay for the moment, since the only use case of faults with runtimes is
+  // by folks using filter level configuration in the fault filter (mainly Lyft folks).
+  if (callbacks_->route() && callbacks_->route()->routeEntry()) {
+    const std::string name = Extensions::HttpFilters::HttpFilterNames::get().FAULT;
+    const auto* route_entry = callbacks_->route()->routeEntry();
+
+    const auto* proto_config =
+        route_entry->perFilterConfig(name) ?: route_entry->virtualHost().perFilterConfig(name);
+
+    if (proto_config) {
+      const envoy::config::filter::http::fault::v2::HTTPFault per_filter_config =
+          dynamic_cast<const envoy::config::filter::http::fault::v2::HTTPFault&>(*proto_config);
+
+      config_.reset(new FaultFilterConfig(per_filter_config, config_->runtime(),
```

Comment:
This is fine, but there would be a perf benefit for using the update callback mechanism that is being added. I would add a TODO to improve this later.